### PR TITLE
fix(embeds): add support for RSC by adding `use client` directives

### PR DESCRIPTION
**Pull request by juliusmarminge**
## What does this PR do?

Fixes # (issue)

Adds `"use client"` directives for embeds so we don't have to make wrapper components when using the embeds in RSCs:

```ts
// app/cal-embed.tsx
"use client";
import _Cal from "@calcom/embed-react";

export const Cal = (props: SomeType) => <_Cal {...props} />;
```

---

Had to make some changes to prettier cause it currently just changed the order and put the `"use client"` at the end of the import section. Part of this was removing an old workaround you had for getting `prettier-plugin-tailwindcss` to work with other plugins. I bumped this version and ran `yarn format` - hence some irrelevant formatting changes are included in this PR too

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by emrysal** [comment]
Nice! Will defer to @hariombalhara for approval but looks excellent

File: packages/config/merged-prettier-plugin.js
Comment (juliusmarminge) [inline packages/config/merged-prettier-plugin.js:1]: This workaround is no longer needed as the tailwind plugin can now be loaded together with other plugins. It was messing with the "use client" directives so fixed it as part of this PR

Comment (juliusmarminge) [reply packages/config/merged-prettier-plugin.js:1]: Thought I'd include "proof" it works - tailwind sorting just fine 👍🏼 


https://user-images.githubusercontent.com/51714798/227368893-de66969e-d46b-452d-b73b-2f8f0936f595.mp4

File: packages/embeds/embed-react/test-cal.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { useEffect } from "react";
-import { useState } from "react";
+import React, { useEffect, useState } from "react";
```
Comment (juliusmarminge) [inline packages/embeds/embed-react/test-cal.tsx:1]: got the standard "react is not part of umd" error here - hence the added `React` import

**Comment by juliusmarminge** [comment]
bump @hariombalhara

**Comment by PeerRich** [comment]
can you fix the merge conflict with yarn.lock @juliusmarminge ?

**Comment by juliusmarminge** [comment]
> can you fix the merge conflict with yarn.lock @juliusmarminge ?

Done!

**Review by hariombalhara — APPROVED** [review]
Looks good to me. Thanks for the change @juliusmarminge

**Pull request merged**